go/vt/wrangler/traffic_switcher.go

```diff
@@ -1015,7 +1025,9 @@ func (ts *trafficSwitcher) changeTableSourceWrites(ctx context.Context, access a
 		}); err != nil {
 			return err
 		}
-		_, err := topotools.RefreshTabletsByShard(ctx, ts.TopoServer(), ts.TabletManagerClient(), source.GetShard(), nil, ts.Logger())
+		rtbsCtx, cancel := context.WithTimeout(ctx, shardTabletRefreshTimeout)
+		defer cancel()
+		_, _, err := topotools.RefreshTabletsByShard(rtbsCtx, ts.TopoServer(), ts.TabletManagerClient(), source.GetShard(), nil, ts.Logger())
```

Comment:
This is a safe/good place to return an error if we could not refresh all tablets as its early on in the operation.

Reply:
sgtm

Reply:
Done: https://github.com/vitessio/vitess/pull/10058/commits/90f656fdb3c0ef4f3f60afa464f6b536ded5712b